ext/gd/gd.c

```diff
@@ -3254,6 +3254,63 @@ PHP_FUNCTION(imagecopyresized)
 }
 /* }}} */
 
+#define GET_TRUECOLOR_PIXEL(px, im, x, y) \
+	if (im->trueColor) { \
+		px = gdImageTrueColorPixel(im, x, y); \
+	} \
+	else { \
+		int c = gdImagePalettePixel(im, x, y); \
+		px = gdTrueColor( \
+			gdImageRed(im, c), \
+			gdImageGreen(im, c), \
+			gdImageBlue(im, c)); \
+	}
+
+/* {{{ Check whether two images are pixel by pixel identical */
+PHP_FUNCTION(imagematch)
+{
+	zval *IM1, *IM2;
+	gdImagePtr im1, im2;
+	int x, y, sx, sy;
+	bool result = true;
+
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_OBJECT_OF_CLASS(IM1, gd_image_ce)
+		Z_PARAM_OBJECT_OF_CLASS(IM2, gd_image_ce)
+	ZEND_PARSE_PARAMETERS_END();
+
+	im1 = php_gd_libgdimageptr_from_zval_p(IM1);
+	im2 = php_gd_libgdimageptr_from_zval_p(IM2);
+
+	sx = im1->sx;
```

Comment:
Probably better to use `gdImageSX()` and `gdImageSY()`.

https://github.com/php/php-src/blob/6fb79e269b926c1d0c4a116c6c5b00b377729547/ext/gd/libgd/gd.h#L768-L769